src/vmm/src/persist.rs

```diff
@@ -247,6 +251,38 @@ fn snapshot_memory_to_file(
     }
 }
 
+/// Validates that snapshot CPU vendor matches the host CPU vendor.
+#[cfg(target_arch = "x86_64")]
+pub fn validate_cpu_vendor(
+    microvm_state: &MicrovmState,
+) -> std::result::Result<(), LoadSnapshotError> {
+    let host_vendor_id = get_vendor_id().map_err(|_| {
+        LoadSnapshotError::CpuVendorMismatch("Failed to read vendor from CPUID.".to_owned())
+    })?;
+
+    // Search for vendor id entry in state CPUID.
+    for entry in microvm_state.vcpu_states[0].cpuid.as_slice().iter() {
+        if entry.function == 0 && entry.index == 0 {
+            let cpu_vendor_id: [u8; 12] =
+                unsafe { std::mem::transmute([entry.ebx, entry.edx, entry.ecx]) };
```

Comment:
This looks somewhat brittle to me, but since we have the integ/unit-tests any endianness/casting/transmuting problems will be caught there so I'm fine with it.

Reply:
This is the same code used in get_vendor_id(). It is safe since we are transmuting an array of [u32; 3]  to [u8; 12].